source/extensions/filters/udp/dns_filter/dns_filter.h

```diff
@@ -26,23 +31,33 @@ namespace DnsFilter {
   COUNTER(answers_aaaa_record)
 
 /**
- * Struct definition for all Dns Filter stats. @see stats_macros.h
+ * Struct definition for all DNS Filter stats. @see stats_macros.h
  */
 struct DnsFilterStats {
   ALL_DNS_FILTER_STATS(GENERATE_COUNTER_STRUCT)
 };
 
-using DnsAddressList = std::vector<std::string>;
-using DnsVirtualDomainConfig = absl::flat_hash_map<std::string, DnsAddressList>;
+using DnsVirtualDomainConfig = absl::flat_hash_map<std::string, AddressConstPtrVec>;
 
+/**
+ * DnsFilter configuration class abstracting access to data necessary for the filter's operation
+ */
 class DnsFilterEnvoyConfig {
 public:
   DnsFilterEnvoyConfig(
       Server::Configuration::ListenerFactoryContext& context,
-      const envoy::config::filter::udp::dns_filter::v2alpha::DnsFilterConfig& config);
+      const envoy::extensions::filters::udp::dns_filter::v3alpha::DnsFilterConfig& config);
 
   DnsFilterStats& stats() const { return stats_; }
-  DnsVirtualDomainConfig& domains() const { return virtual_domains_; }
+  const DnsVirtualDomainConfig& domains() const { return virtual_domains_; }
+  const std::vector<Matchers::StringMatcherPtr>& knownSuffixes() const { return known_suffixes_; }
+  const absl::flat_hash_map<std::string, uint64_t>& domainTtl() const { return domain_ttl_; }
+  const AddressConstPtrVec& resolvers() const { return resolvers_; }
+  bool forwardQueries() const { return forward_queries_; }
+  const std::chrono::milliseconds resolverTimeout() const { return resolver_timeout_ms_; }
+
+  static constexpr uint64_t DefaultResolverTimeoutMs = 500;
+  static constexpr uint64_t DefaultResolverTTLs = 300;
```

Comment:
Please use std::chrono types here and elsewhere for more clarity, then you can drop "Ms", "s", etc.

Reply:
Updated